src/contexts/globalLayout/GlobalNavbar/ProfileDropdown/ProfileDropdownContent.tsx

```diff
@@ -56,32 +58,61 @@ export function ProfileDropdownContent({ showDropdown, onClose, queryRef }: Prop
 
         ...getEditGalleryUrlFragment
         ...ManageWalletsModalFragment
+        ...ManageWalletModalWithEmailFragment
         ...isFeatureEnabledFragment
       }
     `,
     queryRef
   );
 
-  const { showModal } = useModalActions();
+  const { showModal, hideModal } = useModalActions();
   const { handleLogout } = useAuthActions();
 
   const isMobile = useIsMobileWindowWidth();
+  const isEmailFeatureEnabled = isFeatureEnabled(FeatureFlag.EMAIL, query);
+
+  const notificationModalActions = useMemo(() => {
+    const handleSettingsClick = () => {
+      // Hide notification modal
+      hideModal();
+
+      showModal({
+        content: <ManageWalletsModalWithEmail queryRef={query} />,
+        headerText: 'Settings',
+      });
+    };
+
+    return (
+      <StyledCogButton onClick={handleSettingsClick}>
+        <CogIcon />
+      </StyledCogButton>
+    );
+  }, [hideModal, showModal, query]);
 
   const handleNotificationsClick = useCallback(() => {
     showModal({
       content: <NotificationsModal fullscreen={isMobile} />,
       isFullPage: isMobile,
       isPaddingDisabled: true,
       headerVariant: 'standard',
+      headerActions: isEmailFeatureEnabled ? notificationModalActions : false,
```

Comment:
instead of making `showModal` take a one off prop, can we modify the one off `NotificationsModal` to show the right stuff? this way, we don't keep bloating the `showModal` function with more options.

Reply:
Make sense! Move it to its own hooks/component.